awx/main/tasks.py

```diff
@@ -905,6 +905,51 @@ def build_params_process_isolation(self, instance, private_data_dir, cwd):
                 process_isolation_params['process_isolation_ro_paths'].append(instance.ansible_virtualenv_path)
         return process_isolation_params
 
+    def build_params_resource_profiling(self, instance):
+        resource_profiling_params = {}
+        if self.should_use_resource_profiling(instance):
+            base_cgroup = getattr(settings, 'AWX_RESOURCE_PROFILING_BASE_CGROUP', 'ansible-runner')
+            cpu_poll_interval = getattr(settings, 'AWX_RESOURCE_PROFILING_CPU_POLL_INTERVAL', '0.25')
+            mem_poll_interval = getattr(settings, 'AWX_RESOURCE_PROFILING_MEMORY_POLL_INTERVAL', '0.25')
+            pid_poll_interval = getattr(settings, 'AWX_RESOURCE_PROFILING_PID_POLL_INTERVAL', '0.25')
+
+            # Ensure base playbook_profiling dir exists
+            results_dir = '/var/log/tower/playbook_profiling'
```

Comment:
@jladdjr @one-t as discussed, to support iso node collection, lets:

1.  Instruct runner to write this to: `/tmp/awx_<nonsense>/artifacts/<job id>/somewhere-here/`
2. Lose the explicit bwrap whitelist below (we no longer need it)
3. In the `final_run_hook` code (here: https://github.com/ansible/awx/pull/4716/files#diff-9d4ea1dd908b35fb92eaede4bd10bb46R923), look for the *presence* of the perf data artifact from #1 and use something like `shutil.copy_tree` to *copy* the data into `/var/log/tower/playbook_profiling`.

In this way, the perf data on iso nodes will be synced _back_ to controllers and copied _on the controller host_ into the place where its sosreport will find it.

Reply:
Made these changes - deploying cluster w/ the changes to make sure everything works